packages/cursorless-engine/src/snippets/updateSnippetBody.ts

```diff
@@ -0,0 +1,41 @@
+import { getLeadingWhitespace } from "../util/regex";
+
+export function updateSnippetBody(body: string): string {
+  // If snippet body containins tabs don't touch anything.
+  if (body.includes("\t")) {
+    return body;
+  }
+
+  const lines: { indentation: string; rest: string }[] = [];
+  let smallestIndentation: string | undefined = undefined;
+
+  body.split(/\r?\n/).forEach((line) => {
+    const indentation = getLeadingWhitespace(line);
+
+    // Keep track of smallest non empty indentation
+    if (
+      indentation.length > 0 &&
+      (smallestIndentation == null ||
+        indentation.length < smallestIndentation.length)
+    ) {
+      smallestIndentation = indentation;
+    }
+
+    lines.push({ indentation, rest: line.slice(indentation.length) });
+  });
+
+  // No indentation found in snippet body. No change.
+  if (smallestIndentation == null) {
+    return body;
+  }
+
+  return lines
+    .map((line) => {
+      const indentation = line.indentation.replaceAll(
+        smallestIndentation!,
+        "\t",
+      );
```

Comment:
This is assuming perfect multiples of `smallestIndentation`. If that assumption is broken, we will mix spaces and tabs. Is that what we want?

Reply:
Yes. I have added a test for that use case.

Reply:
Why is that desirable?

Reply:
Just because there really is no correct answer. We can round down, round up or leave the extra spaces untouched. I think the latter is the least wrong thing to do.

Reply:
Fwiw in my generate snippet code I just consider whether it's an increase or decrease relative to the previous line, regardless of how much. I believe that's what Python does, but I could be wrong about that

Reply:
Not sure I agree. Mixed tabs and spaces is a very bad failure mode imo. Maybe let's discuss?

Reply:
Lets